front50-core/src/main/java/com/netflix/spinnaker/front50/jackson/TriggerDeserializer.java

```diff
@@ -0,0 +1,18 @@
+package com.netflix.spinnaker.front50.jackson;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.front50.api.model.pipeline.Pipeline;
+import com.netflix.spinnaker.front50.api.model.pipeline.Trigger;
+import java.util.Collection;
+
+public class TriggerDeserializer {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  public static Collection<Trigger> deserialize(Pipeline pipeline) {
+    Collection<Trigger> triggers = pipeline.getTriggers();
```

Comment:
This part is confusing. It uses an object mapper to map `Collection<Trigger>` to `Collection<Trigger>`, which is a no-op.

I'm not sure about Go, but in a language like TypeScript, a cast (`as` in TS) is purely a compile-time check. 

In Java, the `(Collection<Trigger>) super.getOrDefault("triggers", new ArrayList<>())` is both a compile-time check and a runtime check and will fail if `super.getOrDefault("triggers", new ArrayList<>())` isn't actually a collection of triggers. So you'd never really get past this line of code if you weren't already dealing with `Collection<Trigger>`. (At least I think so - please let me know if I'm misinterpreting this).

I would want to know if there's any way that `(Collection<Trigger>) super.getOrDefault("triggers", new ArrayList<>())` can fail. Is there any way to get something under the key `triggers` in the hashmap that isn't a `Collection<Trigger>`? It looks like there's a setter here: https://github.com/mochacat/front50/pull/1/files#diff-0b2bc300fd3965c64ba4184955384aa8609cace04e1c015b31ef4a83552e53d1R93. You can only pass `Collection<Trigger>` in there, so there's no risk in pulling it back out if that's the only way to pass `triggers` into the hashmap. But is there another way? Why was the original code written with an object mapper?

Reply:
Removed deserializer since we found a way around it.